tests/ttnn/unit_tests/operations/test_moreh_adam.py

```diff
@@ -141,3 +144,25 @@ def forward(self, x):
         logger.debug(f"Out passing (max_exp_avg_sq)={passing}")
         logger.debug(f"Output pcc={out}")
     assert passing
+
+
+@skip_for_grayskull("Doesn't seem to work properly on Grayskull devices. Wormhole_b0 devices work fine.")
+@pytest.mark.parametrize(
+    "params",
+    (
+        # shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en
+        ([32, 32], 0.0, (0.9, 0.999), 1e-06, 0.0, True, True),
+        ([2, 2, 2, 2, 2, 2, 64, 64], 0.0, (0.9, 0.999), 1e-06, 0.0, False, False),
+    ),
+)
+def test_moreh_adam_enable_cache(params, device, use_program_cache):
+    for i in range(4):
+        if i % 2 == 1:
+            param_list = list(params)
+            param_list[6] = False if param_list[6] else True
+            params = tuple(param_list)
```

Comment:
Why do we have to change the value of `fp32_dest_acc_en` to `True` and `False` only inside the for loop? Is there a specific reason for implementing it this way?

The `FP32_DEST_ACC_EN` affects the define macro value of the compute kernel, so if this value changes, it doesn't seem likely to result in a cache hit.

Reply:
I've made the necessary changes